src/main/java/org/jenkinsci/plugins/mesos/listener/MesosRunListener.java

```diff
@@ -44,7 +47,13 @@ public void onStarted(Run r, TaskListener listener) {
         try {
           Computer computer = node.toComputer();
           if (computer != null) {
-            String hostname = computer.getHostName();
+            EnvVars environment = computer.getEnvironment();
+            String hostname;
+            if(isBuildRunningInsideContainer(environment)) {
```

Comment:
Instead of depending on the env variable can you do `node.getSlaveInfo().getContainerInfo().type() == DOCKER`?